### PR TITLE
New recipe: quicktour v0.2

### PR DESCRIPTION
**Pull request by humdingerb**
"A quick tour of Haiku, aimed at new users"

### CODE REVIEW

File: haiku-misc/quicktour/quicktour-0.2.recipe [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+widgets and the Deskbar to window management like Stack & Tile and where \
+to get more software. All in under 20 slides.
+After installation a link 'QuickTour' is created on the Desktop."
+HOMEPAGE="https://github.com/humdingerb/quicktour"
+COPYRIGHT="2018 Humdinger"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/humdingerb/quicktour/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="1d99c2efb0e1d466af009e4e079249fbe78cce8cce2335814f3098b632b7be15"
+ADDITIONAL_FILES="quicktour-link-on-desktop.sh"
```
Comment (fbrosson) [inline haiku-misc/quicktour/quicktour-0.2.recipe]: Suggestion:
```diff
 CHECKSUM_SHA256="1d99c2efb0e1d466af009e4e079249fbe78cce8cce2335814f3098b632b7be15"
+SOURCE_FILENAME="quicktour-$portVersion.tar.gz"
 ADDITIONAL_FILES="quicktour-link-on-desktop.sh"
```
[HaikuPorter-Guidelines#archive-name](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#archive-name)

File: haiku-misc/quicktour/quicktour-0.2.recipe [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	quickDir=$documentationDir/quicktour
+	mkdir -p $quickDir
+
+	cp -r en $quickDir
+	cp -r images $quickDir
+	cp -a Haiku-tour.css $quickDir
+
+	mkdir -p $postInstallDir
+	cp $portDir/additional-files/quicktour-link-on-desktop.sh $postInstallDir
+	chmod +x $postInstallDir/quicktour-link-on-desktop.sh
```
Comment (fbrosson) [inline haiku-misc/quicktour/quicktour-0.2.recipe]: Maybe:
```diff
- 	chmod +x $postInstallDir/quicktour-link-on-desktop.sh
```
Since git is able to store the file permissions, you can `chmod +x haiku-misc/quicktour/additional-files/quicktour-link-on-desktop.sh` and remove the `chmod` in `INSTALL`.

I guess haikuporter will preserve it but I did not check this.

Now, I don't know if this change is desirable.

File: haiku-misc/quicktour/additional-files/quicktour-link-on-desktop.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+# Script to create a symlink of the Haiku Quick Tour
+# start page on the Desktop
+
+startPage="01-start.html"
+desktopDir="$(finddir B_DESKTOP_DIRECTORY)"
+
+systemDocDir="$(finddir B_SYSTEM_DOCUMENTATION_DIRECTORY)/quicktour/en"
+userDocDir="$(finddir B_USER_DOCUMENTATION_DIRECTORY)/quicktour/en"
```
Comment (alaviss) [inline haiku-misc/quicktour/additional-files/quicktour-link-on-desktop.sh]: I think we could fetch the output of `findpaths -e B_FIND_PATH_DOCUMENTATION_DIRECTORY quicktour/en/$startPage` instead then links the first entry :)

File: haiku-misc/quicktour/additional-files/quicktour-link-on-desktop.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+mkdir -p $settingsDir
```
Comment (alaviss) [inline haiku-misc/quicktour/additional-files/quicktour-link-on-desktop.sh]: Why is this here?

File: haiku-misc/quicktour/quicktour-0.2.recipe (hunk 1 above)
Comment (humdingerb) [reply haiku-misc/quicktour/quicktour-0.2.recipe]: Done. Thanks!

**Comment by humdingerb** [comment]
Thanks for the review. That `findpaths -e B_FIND_PATH_DOCUMENTATION_DIRECTORY` is really clever. :)

**Comment by humdingerb** [comment]
Am I correct to suspect you guys want me to merge myself? :)

**Comment by fbrosson** [comment]
I've just built the recipe. (Everything is OK.)

> Am I correct to suspect you guys want me to merge myself? :)

It's as you wish :)
(I guess we're a bit lazy :)

BTW, if you merge it yourself you might wish to change the commit message to
```
QuickTour: new recipe
```
(But your current commit title is also OK if you prefer it that way.)

**Comment by humdingerb** [comment]
Done. Thanks!

**Comment by fbrosson** [comment]
@humdingerb, for your next merges, if the only change is the commit message then there's something you can do to avoid replacing the commit with a new one: you can edit the commit message in GitHub's web interface **_right after_** you click on "Squash and merge" (and **_before_** "Confirm merge". )
Well, all methods are OK.

BTW, it seems you used "Rebase and merge". That's also OK, but this does not allow you to edit the commit message, and it does not automatically append the ` (#3182)` at the first line of the commit message.
But don't worry. Not having the number of the PR is not a problem.

**Comment by humdingerb** [comment]
OK. I thought one only uses "Squash & merge" when there are many accumulated commits in  a PR. Otherwise "Rebase"...
